tensorflow_serving/batching/batching_util_test.cc

```diff
@@ -0,0 +1,93 @@
+/* Copyright 2017 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow_serving/batching/batching_util.h"
+
+
+
+#include <gtest/gtest.h>
+#include "tensorflow/core/framework/tensor.h"
+#include "tensorflow/core/lib/core/status.h"
+#include "tensorflow/core/platform/types.h"
+
+namespace tensorflow {
+namespace serving {
+namespace {
+std::vector<std::pair<string, Tensor>> CreateInputsWithTensorShapes(
+        const std::vector<TensorShape>& shapes) {
+  std::vector<std::pair<string, Tensor>> inputs;
+  for (int i = 0; i < shapes.size(); ++i) {
+    inputs.push_back({"x" + std::to_string(i), Tensor(DT_FLOAT, shapes[i])});
+  }
+  return inputs;
+}
+
+
+TEST(BatchingUtilTest, CalculateMaxDimSizes) {
+  std::vector<TensorShape> shapes1;
```

Comment:
Rather than using distinct variable names (shapes1, shapes2, ...) it would be better to use scopes, as in:

{
  shapes = ...;
  EXPECT_...(shapes ...);
}
{
  shapes = ...;
  ...and so on
}

And similarly elsewhere in your test code please.

Reply:
As I understand scopes are useful when there are multiple asserts, but in this case inputs1 and inputs2 are used to form a batch that is used in only one assert. So, I think I can't  use scopes in this test case. If I find other cases where I can use scopes I'll do that.